hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java

```diff
@@ -236,7 +237,9 @@
 
 	private final Map uniqueKeyLoaders = new HashMap();
 	private final Map lockers = new HashMap();
-	private final Map loaders = new HashMap();
+	private UniqueEntityLoader noneLockLoader;
+	private UniqueEntityLoader readLockLoader;
+	private final Map<Object, UniqueEntityLoader> loaders = new ConcurrentHashMap<>();
```

Comment:
So the `Object` here is weird because apparently in the `loaders`, we put  loaders with a `LockMode` key but also with a `String` key: at least we try to get some using a `String` key. Not sure if it is a bug, decided to be on the safe side for this patch.

See https://github.com/hibernate/hibernate-orm/blob/master/hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java#L4260 .

Reply:
This is correct